### ISSUE TITLE
react/jsx-no-comment-textnodes: `{/* eslint-disable-next-line */}` JSX comment directive not matched to the next-line diagnostic — error emitted and directive reported unused

### ISSUE THREAD

**Issue by ahmedelgabri** [oxc-project/oxc#25634]
## Summary

oxlint fails to bind a `{/* eslint-disable-next-line <rule> */}` directive (the JSX expression-container comment form) to a `react/jsx-no-comment-textnodes` diagnostic that it anchors on the immediately following line. It simultaneously emits the live error AND reports the directive as "Unused eslint-disable directive (no problems were reported)". ESLint (7 and 9) treats the directive as used and suppresses the error.

## oxlint version

1.78.0; originally observed on 1.69.0, re-verified on 1.72.0 and 1.78.0.

## Reproduction

`Repro.tsx` (tab-indented):

```tsx
import {type FC} from 'react'

export const Repro: FC<{uri: string}> = ({uri}) => {
	const [protocol, ...url] = uri.split('//')
	return (
		<code>
			{/* eslint-disable-next-line react/jsx-no-comment-textnodes */}
			{protocol}
			//
			{url.join('//')}
		</code>
	)
}
```

`.oxlintrc.json`:

```json
{
	"plugins": ["react"],
	"categories": {"correctness": "off"},
	"settings": {"react": {"version": "18.3"}},
	"rules": {"react/jsx-no-comment-textnodes": "error"}
}
```

```
oxlint -c .oxlintrc.json --report-unused-disable-directives Repro.tsx
```

## Expected

The rule fires on this construct (the `//` text node read as a comment-like child of `<code>`). The `eslint-disable-next-line` on line 7 targets line 8, where oxlint anchors the diagnostic, so the directive should be matched and the error suppressed — zero diagnostics, no "unused directive" warning. ESLint 7 and 9 both fire the rule when the directive is removed, and report nothing (exit 0) when it is present.

## Actual (oxlint 1.78.0)

```
Repro.tsx:8:14: error react(jsx-no-comment-textnodes): Comments inside children section of tag should be placed inside braces
Repro.tsx:7:5: warning: Unused eslint-disable directive (no problems were reported).
```

A variant pins down the mis-anchoring: placing `{/* eslint-disable-line react/jsx-no-comment-textnodes */}` on its own line between `{protocol}` and `//` anchors the diagnostic at the end of the JSX-comment line itself (`8:62`) rather than at the offending text node, and again reports the directive unused. The diagnostic span used for directive matching appears to be computed from the wrong node (the JSX expression-container comment, not the JSXText), breaking line binding specifically for the JSX-comment directive form.

The rule's detection logic is correct (with no directive, the error fires at `7:14` as expected); this is a disable-directive matching defect. Running the real eslint-plugin-react rule via oxlint's jsPlugin path showed identical behavior (confirmed on 1.69.0), ruling out the native rule implementation and localizing the bug to host-side directive binding.

## Cross-checked

| Tool | Versions | Result on `Repro.tsx` (with `--report-unused-disable-directives`) |
| --- | --- | --- |
| ESLint 7.23.0 | @typescript-eslint/parser 6.21.0, eslint-plugin-react 7.37.5 | exit 0 — directive suppresses the error, not reported unused |
| ESLint 9.39.4 (flat config) | typescript-eslint 8.62.1, eslint-plugin-react 7.37.5 | exit 0 — same |
| oxlint 1.78.0 | native `react/jsx-no-comment-textnodes` | error at 8:14 + directive at 7:5 reported unused |
